src/Symfony/Component/Asset/composer.json

```diff
@@ -16,7 +16,9 @@
         }
     ],
     "require": {
-        "php": ">=8.1"
+        "php": ">=8.1",
+        "symfony/filesystem": "^5.4|^6.0",
+        "symfony/mime": "^5.4|^6.0"
```

Comment:
those should be optional dependencies, as only the asset pipeline needs them, which is an optional feature of the component (projects using webpack won't use the asset pipeline).

Alternatively, the AssetPipeline might be big enough to become its own component (depending on symfony/asset), with the additional benefit that FrameworkBundle could then have a smart default to enable it when the package is installed. Projects will generally need either the AssetPipeline or WebpackEncoreBundle but not both

Reply:
Yup - I'd be interested in what others think about this. If we keep this in `Asset`, but make these optional (which is definitely more correct), then users might get met with extra `Run "composer require symfony/mime" to use the pipeline` type of messages. Unless we *also* add an "asset pipeline pack" that contains all of it... but that doesn't make me super excited either.

If I had to lean towards something, it would probably be to split this into a new `AssetPipeline` component.

Reply:
I would vote for a new component as well.

Reply:
if the split is just motivated by the extra deps, I think mime and filesystem are too niche in the new pipeline system to be a string enough reason:

here is the only place where mime is used:
`return self::EXTENSIONS_MAP[$extension] ?? $this->mimeTypes->guessMimeType($filePath);` 

Given the list in `EXTENSIONS_MAP`, this means mime is almost never going to be used in practice. Instead of adding the dep to it, I'd rather suggest to throw an exception when an extenstion is unknown, and add a way to configure the mapping.

For filesystem, we're only using `isAbsolutePath`, so either we borrow some logic from there, our we're fine with the dep. Borrowing might be my preference.

I'd prefer having all this in the Asset component personally.

Reply:
It is not only about the dependencies. It would also allow FrameworkBundle to enable the pipeline automatically based on the installed packages.

Reply:
And we could ship a recipe with a config file that adds your assets/ as a pipeline path. That recipe flexibility might be another reason - though I still need to think about how we’ll accomplish the mixture of the pipeline recipe, importmaps and stimulus recipes… as each system can technically be used without the others 😀